src/go/k8s/pkg/resources/resource.go

```diff
@@ -85,14 +86,29 @@ func CreateIfNotExists(
 	if err := patch.DefaultAnnotator.SetLastAppliedAnnotation(obj); err != nil {
 		return false, fmt.Errorf("unable to add last applied annotation to %s: %w", obj.GetObjectKind().GroupVersionKind().Kind, err)
 	}
-	err := c.Create(ctx, obj)
-	if err != nil && !errors.IsAlreadyExists(err) {
-		return false, fmt.Errorf("unable to create %s resource: %w", obj.GetObjectKind().GroupVersionKind().Kind, err)
+	// this is needed because we cannot pass obj into get as the client methods
+	// modify the input received and we don't want to mutate the object passed
+	// in
+	actual := &unstructured.Unstructured{}
+	actual.SetGroupVersionKind(gvk)
+	err := c.Get(ctx, types.NamespacedName{
+		Namespace: obj.GetNamespace(),
+		Name:      obj.GetName(),
+	}, actual)
+	if err != nil && !errors.IsNotFound(err) {
+		return false, fmt.Errorf("error while fetching %s resource: %w", obj.GetName(), err)
 	}
-	if err == nil {
-		l.Info(fmt.Sprintf("%s %s did not exist, was created", gvk.Kind, obj.GetName()))
-		return true, nil
+	if errors.IsNotFound(err) {
+		err = c.Create(ctx, obj)
+		if err != nil && !errors.IsAlreadyExists(err) {
+			return false, fmt.Errorf("unable to create %s resource: %w", obj.GetObjectKind().GroupVersionKind().Kind, err)
+		}
+		if err == nil {
+			l.Info(fmt.Sprintf("%s %s did not exist, was created", gvk.Kind, obj.GetName()))
+			return true, nil
+		}
 	}
+	// not exists, going to create it
```

Comment:
This particular comment I don't understand as create was done in https://github.com/vectorizedio/redpanda/pull/3268/commits/0c21062f7ac8b06369a3a77ebe3d56bc2fe21a70#diff-add8bf816971d3e4c76f469321f420a65afa670f817e61c516bcf769afe09e07R102

The return statement mean object is not created because it exist.

Reply:
I think it's misplaced.